scripts/dev/upstream_merge/utils/git_commands.py

```diff
@@ -262,6 +275,104 @@ def git_pull_request(
     return run_command(command, capture_output)
 
 
+def git_reset(
+    target="HEAD",
+    hard=False,
+    soft=False,
+    capture_output=True
+):
+    """
+    Reset the repository to a specific commit.
+    :param target: Target commit/branch to reset to (default: "HEAD").
+    :param hard: Whether to do a hard reset (discard all changes).
+    :param soft: Whether to do a soft reset (keep changes staged).
+    :param capture_output: Whether to capture the command output.
+    """
+    command = f"git reset"
+    if hard:
+        command += " --hard"
+    elif soft:
+        command += " --soft"
+    command += f" {target}"
+    return run_command(command, capture_output)
+
+
+def git_clean(
+    force=False,
+    directories=False,
+    ignored_files=False,
+    capture_output=True
+):
+    """
+    Remove untracked files from the working tree.
+    :param force: Force removal of files.
+    :param directories: Remove untracked directories.
+    :param ignored_files: Remove ignored files as well.
+    :param capture_output: Whether to capture the command output.
+    """
+    command = "git clean"
+    if force:
+        command += " -f"
+    if directories:
+        command += " -d"
+    if ignored_files:
+        command += " -x"
+    return run_command(command, capture_output)
+
+
+def git_merge_abort(
```

Comment:
Why not modify the existing git_merge function instead of adding a new one? That way we have lesser number of merge related functions to keep track of.